### PR TITLE
Rename torch_mlir.compile APIs and introduce FX based analogs

### PR DESCRIPTION
**Pull request by saienduri**
Link to related RFC: https://discourse.llvm.org/t/rfc-rename-torch-mlir-compile-apis-and-introduce-fx-based-analogs/76646
This commit updates the documentation, tests, CMake files, and API for the proposed changes in the RFC. There is a new torch_mlir/fx.py for user level APIs related to importing modules and a corresponding test for this path can be found at test/python/fx_importer/basic_test.py.

### CODE REVIEW

**Review by stellaraccident — COMMENTED** [review]
Thanks. This looks pretty good to me. Let's give some more time on the RFC and land this on Thursday or Friday if no objections.

Would be good to get approvals from others impacted for an API break like this.

File: test/python/fx_importer/basic_test.py [hunk 1]
```diff
@@ -13,26 +13,7 @@
 import torch.export
 import torch.nn as nn
```
Comment (aartbik) [inline test/python/fx_importer/basic_test.py:15]: This was pre-existing, but the copyright notice in this file seems inconsistent with the rest of the project (AMD line). Is that intentional, or an oversight?

File: python/torch_mlir/fx.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import torch
+import torch.export
+import torch.nn as nn
+
+from torch_mlir.extras.fx_importer import FxImporter
+from torch_mlir import ir
+from torch_mlir.dialects import torch as torch_d
+
+def export_and_import(
```
Comment (aartbik) [inline python/torch_mlir/fx.py:11]: much better place for this!

File: docs/roadmap.md [hunk 3]
```diff
@@ -51,6 +51,22 @@ the ecosystem are:
...
+  (torch_mlir.torchscript.compile). This allows users to test the compiler's
+  output to the different MLIR dialects (`TORCH`, `TOSA`, `LINALG_ON_TENSORS`,
+  `RAW` and `STABLEHLO`). This path is deprecated and doesn’t give access to
+  the current generation work that is being driven via the fx_importer. It is
+  tied to the old Torchscript path.
+- The second path (torch_mlir.fx.export_and_import) allows users to import a
+  consolidated torch.export.ExportedProgram instance of an arbitrary Python
+  callable (an nn.Module, a function or a method) and output to torch dialect
+  mlir module. This path is what users are using currently, but the path is
+  not quite ready yet.
```
Comment (aartbik) [inline docs/roadmap.md:68]: not quite ready yet -> not fully functional yet?

**Review by aartbik — APPROVED** [review]
This new structure makes the two paths so much more clear (and does not make the torchscript look like the main one ;-) Thanks!

I am a newbee, so probably you want to collect a  few more LGTMs ;-)

File: docs/roadmap.md [hunk 4]
```diff
@@ -51,6 +51,22 @@ the ecosystem are:
...
+- The first path is part of the legacy project pt1 code
+  (torch_mlir.torchscript.compile). This allows users to test the compiler's
+  output to the different MLIR dialects (`TORCH`, `TOSA`, `LINALG_ON_TENSORS`,
+  `RAW` and `STABLEHLO`). This path is deprecated and doesn’t give access to
+  the current generation work that is being driven via the fx_importer. It is
+  tied to the old Torchscript path.
+- The second path (torch_mlir.fx.export_and_import) allows users to import a
+  consolidated torch.export.ExportedProgram instance of an arbitrary Python
+  callable (an nn.Module, a function or a method) and output to torch dialect
+  mlir module. This path is what users are using currently, but the path is
```
Comment (ramiro050) [inline docs/roadmap.md:67]: nit: I would replace "This path is what users are using currently" to "This path is aligned with PyTorch's roadmap, ...". I think we currently have many users that still use the old path :)

File: projects/pt1/python/test/compile_api/tracing.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 import torch
 
-import torch_mlir
+import torch_mlir.torchscript as torchscript
```
Comment (ramiro050) [inline projects/pt1/python/test/compile_api/tracing.py:10]: nit: you can do `from torch_mlir import torchscript` to avoid repeating `torchscript`

**Review by ramiro050 — APPROVED** [review]
Thanks for the quick work on this! A couple of small comments

File: python/CMakeLists.txt [hunk 6]
```diff
@@ -37,6 +37,7 @@ declare_mlir_python_sources(TorchMLIRPythonSources.Importers
   SOURCES
     extras/fx_importer.py
     extras/onnx_importer.py
+    fx.py
```
Comment (stellaraccident) [inline python/CMakeLists.txt:40]: I think we want a different group for this. Perhaps "PublicAPI". We're not ready to export this to downstreams yet, which is what this group does.

File: test/python/fx_importer/basic_test.py (hunk 1 above)
Comment (saienduri) [reply test/python/fx_importer/basic_test.py:15]: Not sure, but changed to be consistent.

File: docs/roadmap.md (hunk 4 above)
Comment (saienduri) [reply docs/roadmap.md]: ah okay, makes sense

File: python/CMakeLists.txt (hunk 6 above)
Comment (saienduri) [reply python/CMakeLists.txt:46]: okay created a PublicAPI group.

**Pull request merged**